pkg/cmd/template/schema_test.go

```diff
@@ -2533,6 +2743,141 @@ foo: 0
+    = hint: value must be the same type as the annotated node.
+    = hint: value must be in Starlark format, e.g.: {'key': 'value'}, True.
+`
+
+			filesToProcess := files.NewSortedFiles([]*files.File{
+				files.MustNewFileFromSource(files.NewBytesSource("schema.yml", []byte(schemaYAML))),
+				files.MustNewFileFromSource(files.NewBytesSource("template.yml", []byte(templateYAML))),
+			})
+
+			assertFails(t, filesToProcess, expectedErr, opts)
+		})
+		t.Run("has multiple values", func(t *testing.T) {
+			schemaYAML := `#@data/values-schema
+---
+#@schema/default 1, 2
+foo: 0
+`
+			templateYAML := `#@ load("@ytt:data", "data")
+---
+foo: #@ data.values.foo
+`
+			expectedErr := `
+Invalid schema
+==============
+expected @schema/default annotation to contain one argument as default value
+
+schema.yml:
+    |
+  4 | foo: 0
+    |
+
+    = found: 2 values
+    = expected: integer (by schema.yml:4)
+    = hint: value must be the same type as the annotated node.
+    = hint: value must be in Starlark format, e.g.: {'key': 'value'}, True.
+`
+
+			filesToProcess := files.NewSortedFiles([]*files.File{
+				files.MustNewFileFromSource(files.NewBytesSource("schema.yml", []byte(schemaYAML))),
+				files.MustNewFileFromSource(files.NewBytesSource("template.yml", []byte(templateYAML))),
+			})
+
+			assertFails(t, filesToProcess, expectedErr, opts)
+		})
+		t.Run("value is an invalid starlark Tuple", func(t *testing.T) {
+			schemaYAML := `#@data/values-schema
+---
+#@schema/default any=True
+foo: 0
+`
+			templateYAML := `#@ load("@ytt:data", "data")
+---
+foo: #@ data.values.foo
+`
+			expectedErr := `
+Invalid schema
+==============
+expected @schema/default annotation to contain value with type of annotated node
+
+schema.yml:
+    |
+  4 | foo: 0
+    |
+
+    = found: starlark.Tuple
+    = expected: integer (by schema.yml:4)
+    = hint: value must be the same type as the annotated node.
+    = hint: value must be in Starlark format, e.g.: {'key': 'value'}, True.
+`
+
+			filesToProcess := files.NewSortedFiles([]*files.File{
+				files.MustNewFileFromSource(files.NewBytesSource("schema.yml", []byte(schemaYAML))),
+				files.MustNewFileFromSource(files.NewBytesSource("template.yml", []byte(templateYAML))),
+			})
+
+			assertFails(t, filesToProcess, expectedErr, opts)
+		})
+		t.Run("is on an array item", func(t *testing.T) {
+			schemaYAML := `#@data/values-schema
+---
+foo:
+#@schema/default "baz"
+- bar
+
+`
+			templateYAML := `#@ load("@ytt:data", "data")
+---
+foo: #@ data.values.foo
+`
+			expectedErr := `
+Invalid schema - @schema/default not supported on array item
+============================================================
+
+schema.yml:
+    |
+  5 | - bar
+    |
+
+    = found: @schema/default annotation on array item
+    = expected: @schema/default annotation to be on map item
+    = hint: place annotation on the map item containing this array item
+    = hint: default value should set the value for the array
```

Comment:
Observing:
- with each hint that directs action, we want to avoid misleading the user
- I wonder if we can convey the message in fewer words (reducing the amount of reading the user needs to do)
- technically, an array can be held by _either_ a map item _or_ a document.

Something like...

```suggestion
Invalid schema - @schema/default not supported on array item
============================================================

schema.yml:
    |
  5 | - thing
    |

    = hint: do you meant to set a default value for the array?
    = hint: set an array's default by annotating its parent
```